src/Microsoft.ML.Data/DataLoadSave/PartitionedFileLoader.cs

```diff
@@ -0,0 +1,726 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Data.Conversion;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+
+[assembly: LoadableClass(PartitionedFileLoader.Summary, typeof(PartitionedFileLoader), typeof(PartitionedFileLoader.Arguments), typeof(SignatureDataLoader),
+    PartitionedFileLoader.UserName, PartitionedFileLoader.LoadName, PartitionedFileLoader.ShortName)]
+
+[assembly: LoadableClass(PartitionedFileLoader.Summary, typeof(PartitionedFileLoader), null, typeof(SignatureLoadDataLoader),
+    PartitionedFileLoader.UserName, PartitionedFileLoader.LoadName, PartitionedFileLoader.ShortName)]
+
+namespace Microsoft.ML.Runtime.Data
+{
+    /// <summary>
+    /// Loads a set of directory partitioned files into an IDataView.
+    /// The directories of the file will treated as column data and the underlying files are loaded using the data loader.
+    /// The first file will be used as the basis for all follow-up file paths and schemas. Any files that don't match
+    /// the expected path or schema will be skipped.
+    /// </summary>
+    /// <example>
+    /// Sample directory structure:
+    ///
+    /// Data/
+    ///     Year=2017/
+    ///         Month=01/
+    ///             data1.parquet
```

Comment:
>parquet [](start = 26, length = 7)

this parquet extension is a bit confusing. In your tests you use csv, can we stick to csv instead of parquet?

Reply:
This partitioning scheme is specked around parquet. The tests use CSV simply for readability. WNF.